web/src/components/apps/Dashboard.tsx

```diff
@@ -26,47 +26,136 @@ const COMMON_ERRORS = {
-    uploadingAirgapFile: false,
-    airgapUploadError: null,
-    viewAirgapUploadError: false,
-    viewAirgapUpdateError: false,
-    airgapUpdateError: "",
-    startSnapshotErrorMsg: "",
-    showAutomaticUpdatesModal: false,
-    showAppStatusModal: false,
-    dashboard: {
-      appStatus: null,
-      metrics: [],
-      prometheusAddress: "",
-    },
-    getAppDashboardJob: new Repeater(),
-    fetchAppDownstreamJob: new Repeater(),
-    gettingAppLicenseErrMsg: "",
-    startSnapshotOptions: [
-      { option: "partial", name: "Start a Partial snapshot" },
-      { option: "full", name: "Start a Full snapshot" },
-      { option: "learn", name: "Learn about the difference" },
-    ],
-    selectedSnapshotOption: { option: "full", name: "Start a Full snapshot" },
-    snapshotDifferencesModal: false,
+// Types
+import {
+  App,
+  AppLicense,
+  Downstream,
+  DashboardResponse,
+  DashboardActionLink,
+  KotsParams,
+  ResourceStates,
+  Version,
+} from "@types";
+import { RouteComponentProps } from "react-router-dom";
+
+type Props = {
+  app: App;
+  cluster: {
+    // TODO: figure out if this is actually a "" | number- maybe just go with number
+    id: "" | number;
   };
+  isBundleUploading: boolean;
+  isHelmManaged: boolean;
+  isVeleroInstalled: boolean;
+  makeCurrentVersion: (version: Version) => void;
+  ping: (clusterId?: string) => void;
+  redeployVersion: (version: Version) => void;
+  refreshAppData: () => void;
+  snapshotInProgressApps: string[];
+  toggleIsBundleUploading: (isUploading: boolean) => void;
+  updateCallback: () => void | null;
+} & RouteComponentProps<KotsParams>;
+
+type SnapshotOption = {
+  option: string;
+  name: string;
+};
+
+// TODO:  update these strings so that they are not nullable (maybe just set default to "")
+type State = {
+  activeChart: string | null;
+  airgapUpdateError: string;
+  airgapUploader: AirgapUploader | null;
+  airgapUploadError: string | null;
+  appLicense: AppLicense | {};
+  appName: string;
+  checkingForUpdateError: boolean;
+  checkingForUpdates: boolean;
+  checkingUpdateMessage: string;
+  currentVersion: Version | {};
+  dashboard: DashboardResponse;
+  displayErrorModal: boolean;
+  downstream: Downstream | null;
+  fetchAppDownstreamJob: Repeater;
+  getAppDashboardJob: Repeater;
+  gettingAppErrMsg: string;
+  gettingAppLicenseErrMsg: string;
+  iconUri: string;
+  loadingApp: boolean;
+  links: DashboardActionLink[];
+  noUpdatesAvalable: boolean;
+  selectedSnapshotOption: SnapshotOption;
+  showAppStatusModal: boolean;
+  showAutomaticUpdatesModal: boolean;
+  snapshotDifferencesModal: boolean;
+  startingSnapshot: boolean;
+  startSnapshotErr: boolean;
+  startSnapshotErrorMsg: string;
+  startSnapshotOptions: SnapshotOption[];
+  updateChecker: Repeater;
+  uploadProgress: number;
+  uploadResuming: boolean;
+  uploadSize: number;
+  uploadingAirgapFile: boolean;
+  viewAirgapUpdateError: boolean;
+  viewAirgapUploadError: boolean;
+};
```

Comment:
when we convert this to using functional components instead of class, this will save on a bunch of lines of extra code as we can define the types inline with useState :)